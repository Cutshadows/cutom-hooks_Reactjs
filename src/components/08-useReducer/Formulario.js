import React from 'react';
import PropTypes from 'prop-types'; 
import { useForm } from '../../hooks/useForm';

const Formulario = ({handleAddTodo}) => {
    const [{description}, handleInputChange, reset]=useForm({
        description:''
    });

    const handleSubmit=(e)=>{
        e.preventDefault();
        if(description.trim().length<= 1){
            return;
        }
        const newTodo={
            id: new Date().getTime(),
            desc:description,
            done: false
        };
        handleAddTodo(newTodo);
        reset();
    }
    return ( 
        <form onSubmit={handleSubmit}>
            <input type="text"
                name="description"
                value={description}
                onChange={handleInputChange}
                className="form-control"
                placeholder="Aprender..."
                autoComplete="off"/>
            <button
            type="submit"
            className="btn btn-outline-success mt-1 w-100">Agregar</button>

        </form>
     );
}
Formulario.protoTypes={
    handleAddTodo:PropTypes.func,
}
 
export default Formulario;