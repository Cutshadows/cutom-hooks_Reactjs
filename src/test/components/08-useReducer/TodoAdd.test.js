import { shallow } from 'enzyme';
import { demotTodos } from '../../fixtures/demoTodos';
import Formulario from '../../../components/08-useReducer/Formulario';


describe('Pruebas sobre <TodoAdd></TodoAdd>', () => {
    const handleAddTodo=jest.fn();
    const wrapperTodoAdd=shallow(
    <Formulario 
    handleAddTodo={handleAddTodo}
    />)

    test('deberia de cargar el wrapper ', () => {
        expect(wrapperTodoAdd).toMatchSnapshot();
    });

    test('No debe de llamar handleAddTodo', () => {
        const formSubmit=wrapperTodoAdd.find('form').prop('onSubmit');
        console.log(formSubmit)

        formSubmit({preventDefault(){}}); //otra forma de hacerlo

        expect(handleAddTodo).toHaveBeenCalledTimes(0);
    })
    test('debe de llamar handleAddTodo', () => {
        const inputValue='Aprender React';
        // console.log(wrapperTodoAdd.find('input').html());
        wrapperTodoAdd.find('input').simulate('change', 
        {
            target:{
                name:'description',
                value:inputValue,
            }
        });
        // console.log(wrapperTodoAdd.find('input').html());

        const formSubmit=wrapperTodoAdd.find('form').prop('onSubmit');
        formSubmit({preventDefault(){}});

        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object)); //{}
        expect(handleAddTodo).toHaveBeenCalledWith({
            desc:inputValue,
            done:false,
            id:expect.any(Number),
        }); //{}

        expect(wrapperTodoAdd.find('input').prop('value')).toBe('');
    })


    
})
