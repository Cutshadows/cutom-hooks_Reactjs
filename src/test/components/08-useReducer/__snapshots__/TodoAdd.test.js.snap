// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <TodoAdd></TodoAdd> deberia de cargar el wrapper  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoComplete="off"
    className="form-control"
    name="description"
    onChange={[Function]}
    placeholder="Aprender..."
    type="text"
    value=""
  />
  <button
    className="btn btn-outline-success mt-1 w-100"
    type="submit"
  >
    Agregar
  </button>
</form>
`;
